File: 12_end/03_plumber/testme.R
# testme.R
# Smoke-test the Brussels Plumber prediction endpoint.
# Tim Fraser
# Rscript 12_end/03_plumber/testme.R

# Load packages and vars
library(httr2)
readRenviron(".env")

# Set base URL
base = trimws(Sys.getenv("API_PUBLIC_URL", unset = "http://localhost:8000"))
# Remove trailing slash if present
base = sub("/$", "", base)

# Send request
r = httr2::request(paste0(base, "/predict")) |>
  httr2::req_url_query(day_of_week = 1, hour_of_day = 8) |>
  httr2::req_timeout(30) |>
  httr2::req_perform()

# Print response
cat("status:", httr2::resp_status(r), "\n")
print(httr2::resp_body_json(r, simplifyVector = TRUE))

File: 12_end/03_fastapi/testme.py
# testme.py
# Smoke-test the FastAPI prediction endpoint.
# Tim Fraser
#
# Optional .env:
# - API_PUBLIC_URL (default: http://localhost:8000)
#
# pip install requests python-dotenv

import os
from pathlib import Path

import requests
from dotenv import load_dotenv


def main() -> None:
    here = Path(__file__).resolve().parent
    os.chdir(here)
    load_dotenv()

    base = os.getenv("API_PUBLIC_URL", "http://localhost:8000").rstrip("/")
    url = f"{base}/predict"
    params = {"day_of_week": 1, "hour_of_day": 8}

    response = requests.get(url, params=params, timeout=30)
    response.raise_for_status()

    print("status:", response.status_code)
    print("url:", response.url)
    print("body:", response.json())


if __name__ == "__main__":
    main()
